src/validationinterface.cpp

```diff
@@ -133,47 +137,62 @@ void CMainSignals::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlockInd
     // the chain actually updates. One way to ensure this is for the caller to invoke this signal
     // in the same critical section where the chain is updated
 
-    m_internals->m_schedulerClient.AddToProcessQueue([pindexNew, pindexFork, fInitialDownload, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, pindexNew, pindexFork, fInitialDownload, this] {
+        if (pindexFork) LogPrint(BCLog::VALIDATIONINTERFACE, "%s: fork block hash=%s\n", func, pindexFork->GetBlockHash().ToString());
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: new block hash=%s\n", func, pindexNew->GetBlockHash().ToString());
         m_internals->UpdatedBlockTip(pindexNew, pindexFork, fInitialDownload);
     });
 }
 
 void CMainSignals::TransactionAddedToMempool(const CTransactionRef &ptx) {
-    m_internals->m_schedulerClient.AddToProcessQueue([ptx, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, ptx, this] {
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: tx hash=%s\n", func, ptx->GetHash().ToString());
         m_internals->TransactionAddedToMempool(ptx);
     });
 }
 
 void CMainSignals::TransactionRemovedFromMempool(CTransactionRef ptx, MemPoolRemovalReason reason) {
     if (reason != MemPoolRemovalReason::BLOCK && reason != MemPoolRemovalReason::CONFLICT) {
-        m_internals->m_schedulerClient.AddToProcessQueue([ptx, this] {
+        auto func = __func__;
+        m_internals->m_schedulerClient.AddToProcessQueue([func, ptx, this] {
+            LogPrint(BCLog::VALIDATIONINTERFACE, "%s: tx hash=%s\n", func, ptx->GetHash().ToString());
             m_internals->TransactionRemovedFromMempool(ptx);
         });
     }
 }
 
 void CMainSignals::BlockConnected(const std::shared_ptr<const CBlock> &pblock, const CBlockIndex *pindex, const std::shared_ptr<const std::vector<CTransactionRef>>& pvtxConflicted) {
-    m_internals->m_schedulerClient.AddToProcessQueue([pblock, pindex, pvtxConflicted, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, pblock, pindex, pvtxConflicted, this] {
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: block hash=%s\n", func, pblock->GetHash().ToString());
```

Comment:
Thanks to pindex, maybe log the height too

Reply:
Added in 4bacb54.